sotodlib/site_pipeline/update_smurf_caldbs.py

```diff
@@ -109,15 +131,309 @@ def smurf_detset_info(config:str | dict,
+
+    for i, cal in enumerate(cals):
+        band = am.det_info.smurf.band[i]
+        chan = am.det_info.smurf.channel[i]
+        detset = am.det_info.detset[i]
+        stream_id = am.det_info.stream_id[i]
+        bg = cal.bg
+        bsa = bsas[detset]
+
+        if bsa is None or bg == -1:
+            continue
+
+        bl_label = f'{stream_id}_b{bg:0>2}'
+        if not bias_line_is_valid[bl_label]:
+            continue
+
+        ridx = np.where(
+            (bsa['bands'] == band) & (bsa['channels'] == chan)
+        )[0]
+        if not ridx: # Channel doesn't exist in bias step analysis
+            continue
+
+        ridx = ridx[0]
+        cal.r_tes = bsa['R0'][ridx]
+        cal.r_frac = bsa['Rfrac'][ridx]
+        cal.p_bias = bsa['Pj'][ridx]
+        cal.s_i = bsa['Si'][ridx]
+        cal.phase_to_pW = pA_per_phi0 / (2*np.pi) / cal.s_i * cal.polarity
+
+    rset = core.metadata.ResultSet.from_friend(np.array(
+        [astuple(c) for c in cals], dtype=cal_dtype
+    ))
+    return rset
+
+
+def get_obs_with_detsets(ctx, detset_idx):
+    """Gets all observations with type 'obs' that have detset data"""
+    db = core.metadata.ManifestDb(detset_idx)
+    detsets = db.get_entries(['dataset'])['dataset']
+    obs_ids = set()
+    for dset in detsets:
+        obs_ids = obs_ids.union(ctx.obsfiledb.get_obs_with_detset(dset))
+    return obs_ids
+
+
+def update_det_caldb(ctx, idx_path, detset_idx, h5_path, logger=None, 
+                     show_pb=False, format_exc=False, overwrite=True):
+    if logger is None:
+        logger = default_logger
+
+    if not os.path.exists(idx_path):
+        scheme = core.metadata.ManifestScheme()
+        scheme.add_exact_match('obs:obs_id')
+        scheme.add_data_field('dataset')
+        db = core.metadata.ManifestDb(scheme=scheme)
+        db.to_file(idx_path)
+    db = core.metadata.ManifestDb(idx_path)
+    
+    # detset_db = metadata.Manifest(detset_idx)
+    existing_obsids = set(db.get_entries(['dataset'])['dataset'])
+    obsids_with_detsets = get_obs_with_detsets(ctx, detset_idx)
+    obsids_with_obs = set(ctx.obsdb.query("type=='obs'")['obs_id'])
+    remaining_obsids = obsids_with_detsets.intersection(obsids_with_obs) \
+                        - existing_obsids
+    # Sort remaining obs_ids by timestamp
+    remaining_obsids = sorted(remaining_obsids,
+                              key=(lambda s: s.split('_')[1]))
+
+    logger.info("%d datasets to add", len(remaining_obsids))
+    for obs_id in tqdm(remaining_obsids, disable=(not show_pb)):
+        try:
+            rset = get_cal_resset(ctx, obs_id)
+        except Exception as e:
+            logger.error("Failed on %s: %s", obs_id, e)
+            if format_exc:
+                logger.error(traceback.format_exc())
+            continue
+
+        logger.info("Writing metadata for %s", obs_id)
+        write_dataset(rset, h5_path, obs_id, overwrite=overwrite)
+        db.add_entry({
```

Comment:
this line will throw an error if overwrite=true and the entry already exists.

Reply:
isn't it the opposite, and it would throw an error if overwrite=False?